src/feat/rir-generator.cc

```diff
@@ -0,0 +1,247 @@
+// feat/rir-generator.cc
+
+// Copyright 2018 Jian Wu
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "feat/rir-generator.h"
+
+namespace kaldi {
+
+void RirGenerator::ComputeDerived() {
+    if (!str_to_pattern_.count(opts_.microphone_type))
+        KALDI_ERR << "Unknown option values: --microphone-type=" << opts_.microphone_type;
+    KALDI_ASSERT(frequency_ >= 1);
+    KALDI_ASSERT(velocity_ >= 1);
+    KALDI_ASSERT(order_ >= -1);
+
+    // Process room topo
+    KALDI_ASSERT(opts_.room_topo != "" && "Options --room-topo is not configured");
+    std::vector<BaseFloat> topo;
+    KALDI_ASSERT(SplitStringToFloats(opts_.room_topo, ",", false, &topo));
+    room_dim_ = topo.size();
+    // KALDI_ASSERT(room_dim_ == 2 || room_dim_ == 3);
+    KALDI_ASSERT(room_dim_ == 3);
+    room_topo_.CopyFromVector(topo);
+
+    // Process source location
+    KALDI_ASSERT(opts_.source_location != "" && "Options --source-location is not configured");
+    std::vector<BaseFloat> loc;
+    KALDI_ASSERT(SplitStringToFloats(opts_.source_location, ",", false, &loc));
+    source_location_.CopyFromVector(loc);
+
+
+    // Process receiver_location
+    KALDI_ASSERT(opts_.receiver_location != "" && "Options --receiver-location is not configured");
+    std::vector<std::string> mics;
+    SplitStringToVector(opts_.receiver_location, ";", false, &mics);
+    num_mics_ = mics.size();
+    receiver_location_.resize(num_mics_);
+    for (int32 i = 0; i < num_mics_; i++) {
+        KALDI_ASSERT(SplitStringToFloats(mics[i], ",", false, &loc));
+        receiver_location_[i].CopyFromVector(loc);
+    }
+
+    // Process angle
+    std::vector<BaseFloat> angle_tmp;
+    if (opts_.orientation == "") {
+        for (int32 i = 0; i <= 1; i++)
+            angle_tmp.push_back(0.0);
+    } else {
+        KALDI_ASSERT(SplitStringToFloats(opts_.orientation, ",", false, &angle_tmp));
+    }
+    if (angle_tmp.size() == 1)
+        angle_tmp.push_back(0.0);
+    KALDI_ASSERT(angle_tmp.size() == 2);
+    angle_.swap(angle_tmp);
+
+    // Process beta

```

Comment:
I think this section needs some comments. What is 6 for?

Reply:
beta means T60 or reflection coefficients for each space(3D room), so dim of it is 1 or 6.